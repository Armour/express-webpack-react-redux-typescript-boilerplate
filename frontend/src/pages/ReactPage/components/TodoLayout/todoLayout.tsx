import React from 'react';
import { NamespacesConsumer } from 'react-i18next';

import TodoFooter from './components/TodoFooter';
import TodoInput from './components/TodoInput';
import TodoList from './components/TodoList';

const styles = require('./todoLayout.scss');

export class TodoLayout extends React.Component {
  public render() {
    return (
      <NamespacesConsumer ns='reactPage'>
        {(t) => (
          <div className={`center-align z-depth-2 ${styles['todo-layout']}`}>
            <span className={styles['todo-title']}>{t('title')}</span>
            <TodoInput />
            <TodoList />
            <TodoFooter />
          </div>
        )}
      </NamespacesConsumer>
    );
  }
}

export default TodoLayout;
