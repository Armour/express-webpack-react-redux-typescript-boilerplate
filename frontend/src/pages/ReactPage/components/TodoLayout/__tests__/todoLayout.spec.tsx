import React from 'react';
import TestRenderer from 'react-test-renderer';

import { TodoLayout } from '../todoLayout';

jest.mock('../components/TodoFooter', () => 'TodoFooter');
jest.mock('../components/TodoInput', () => 'TodoInput');
jest.mock('../components/TodoList', () => 'TodoList');

describe('TodoLayout', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <TodoLayout />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
