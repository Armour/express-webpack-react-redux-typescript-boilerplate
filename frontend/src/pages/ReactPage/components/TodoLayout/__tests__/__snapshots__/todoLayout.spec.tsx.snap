// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoLayout should renders correctly 1`] = `
<div
  className="center-align z-depth-2 todo-layout"
>
  <span
    className="todo-title"
  >
    title
  </span>
  <TodoInput />
  <TodoList />
  <TodoFooter />
</div>
`;
